src/lightning/pytorch/strategies/xla.py

```diff
@@ -120,6 +115,9 @@ def setup(self, trainer: "pl.Trainer") -> None:
         self.setup_precision_plugin()
 
         if trainer.state.fn == TrainerFn.FITTING:
+            from torch_xla.experimental import pjrt
+
+            pjrt.broadcast_master_param(self.model)
```

Comment:
added master param broadcasting here.

Not tested yet.

Reply:
Why do this only during `.fit`? I think the rest might need it too.

Also, shouldn't it be done under `pjrt.using_pjrt()`?

Reply:
I'm matching what is done in `DDPStrategy`, but I'm fine to put it outside this condition

This can be used for both XRT and PJRT, so there is no check for `pjrt.using_pjrt()`

Reply:
> I'm matching what is done in DDPStrategy, but I'm fine to put it outside this condition

You made me notice that we have a bug. During non-fit, we don't wrap with DDP because there are no gradients to reduce, but then the parameters are not synchronized.

The equivalent of `pjrt.broadcast_master_params` would be this call: https://github.com/pytorch/pytorch/blob/v2.0.0/torch/nn/parallel/distributed.py#L675-L682

This is not a big issue because during any realistic non-fit use, you'd load a checkpoint, but we might want to fix it anyways. I'm fine keeping this under the condition and addressing this separately.

cc @awaelchli 

> This can be used for both XRT and PJRT, so there is no check for pjrt.using_pjrt()

XRT doesn't broadcast them automatically, does it? If not then we should keep it.

Reply:
> XRT doesn't broadcast them automatically, does it? If not then we should keep it.

Right, this is needed for both XRT and PJRT.